### PR TITLE
Eagly calculate address and password in ProxyProvider equals and hashcode

### PR DESCRIPTION
**Pull request by fred84**
Possible solution for #566

### CODE REVIEW

File: src/test/java/reactor/netty/tcp/ProxyProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public void differentAddresses() {
+        assertNotEquals(createProxy(ADDRESS_1, PASSWORD_1), createProxy(ADDRESS_2, PASSWORD_1));
+        assertNotEquals(createProxy(ADDRESS_1, PASSWORD_1).hashCode(), createProxy(ADDRESS_2, PASSWORD_1).hashCode());
+    }
+
+    @Test
+    public void differentPasswords() {
+        assertNotEquals(createProxy(ADDRESS_1, PASSWORD_2), createProxy(ADDRESS_2, PASSWORD_2));
+        assertNotEquals(createProxy(ADDRESS_1, PASSWORD_2).hashCode(), createProxy(ADDRESS_2, PASSWORD_2).hashCode());
+    }
```
Comment (violetagg) [inline src/test/java/reactor/netty/tcp/ProxyProviderTest.java]: Here the passwords are the same, but the addresses are different. Shouldn't it be opposite?

File: src/main/java/reactor/netty/tcp/ProxyProvider.java [hunk 2]
```diff
@@ -213,15 +213,20 @@ public boolean equals(Object o) {
...
 	@Override
 	public int hashCode() {
-		return Objects.hash(username, password, getAddress(), getNonProxyHosts(), getType());
+		return Objects.hash(username, getPasswordValue(), getAddress().get(), getNonProxyHosts(), getType());
+	}
+
+	@Nullable
+	private String getPasswordValue() {
+		return null == password ? null : password.apply(username);
 	}
```
[the remark this answers is not in the document]
Comment (fred84) [reply src/main/java/reactor/netty/tcp/ProxyProvider.java]: done

File: src/test/java/reactor/netty/tcp/ProxyProviderTest.java (hunk 1 above)
Comment (fred84) [reply src/test/java/reactor/netty/tcp/ProxyProviderTest.java]: fixed

**Comment by violetagg** [comment]
Thanks

**Pull request merged**